libbeat/common/kubernetes/watcher.go

```diff
@@ -123,7 +125,8 @@ func NewWatcher(client kubernetes.Interface, resource Resource, opts WatchOption
 			new, _ := accessor.ResourceVersion(n.(runtime.Object))
 
 			// Only enqueue changes that have a different resource versions to avoid processing resyncs.
-			if old != new {
+			// Also if there is a registered function to check for updates, use it to check for updates.
+			if old != new && opts.IsUpdated != nil && opts.IsUpdated(o, n) {
```

Comment:
`isUpdated` will be nil for all other resources (Pod, Service...), so this will never be true for them.

Perhaps we should have a default `isUpdated` function, when the option is not present that does just the `old != new` check?